lib/GADS/Schema/Result/Report.pm

```diff
@@ -270,31 +286,38 @@ Function to create a PDF of the report - it will return a PDF object
 =cut
 
 sub create_pdf
-{   my ($self, $record) = @_;
-
-    my $dateformat = GADS::Config->instance->dateformat;
-    my $now        = DateTime->now;
-    $now->set_time_zone('Europe/London');
-    my $now_formatted = $now->format_cldr($dateformat) . " at " . $now->hms;
-    my $updated =
-      $self->created->format_cldr($dateformat) . " at " . $self->created->hms;
-
-    my $config = GADS::Config->instance;
-    my $header = $config && $config->gads && $config->gads->{header};
-    my $pdf    = CtrlO::PDF->new(
-        header => $header,
-        footer => "Downloaded by " . $self->user->value . " on $now_formatted",
-    );
-
-    $pdf->add_page;
-    $pdf->heading( $self->name );
-    $pdf->heading( $self->description, size => 14 ) if $self->description;
-
-    my $fields = [ [ 'Field', 'Value' ] ];
+{   my ($self, $record, $default_marking, $logo) = @_;
+
+    my $marking = $self->security_marking || $default_marking;
+
+    my $pdf;
+
+    if($logo) {
+        $pdf    = CtrlO::PDF->new(
+            header => $marking,
+            footer => $marking,
+            logo   => $logo,
+        );
+        $pdf->add_page;
+        my $lines = [];
+        $lines= $self->_wraptext($self->description, 50) if($self->description);
+        my $topmargin = -40 - (scalar(@$lines) * 20);
```

Comment:
With the inclusion of `_wraptext` this allows this to be calculated in order to place the report headers in the relevant place.

Also, due to pulling the headers up, the wrapping of the normal `text` field would overlap with the logo as it's bounds are set within that of the logo.